course-mw/enrolment-actor/src/main/scala/org/sunbird/aggregate/CollectionSummaryAggregate.scala

```diff
@@ -0,0 +1,177 @@
+         |      {
+         |        "type": "or",
+         |        "fields": [
+         |          {
+         |            "type": "selector",
+         |            "dimension": "edata_type",
+         |            "value": "complete"
+         |          },
+         |          {
+         |            "type": "selector",
+         |            "dimension": "edata_type",
+         |            "value": "enrollment"
+         |          },
+         |          {
+         |            "type": "selector",
+         |            "dimension": "edata_type",
+         |            "value": "certificate-issued"
+         |          }
+         |        ]
+         |      },
+         |      {
+         |        "type": "and",
+         |        "fields": [
+         |          {
+         |            "type": "selector",
+         |            "dimension": "context_cdata_id",
+         |            "value": "$batchId"
+         |          },
+         |          {
+         |            "type": "and",
+         |            "fields": [
+         |              {
+         |                "type": "selector",
+         |                "dimension": "object_rollup_l1",
+         |                "value": "$courseId"
+         |              },
+         |              {
+         |                "type": "selector",
+         |                "dimension": "eid",
+         |                "value": "AUDIT"
+         |              }
+         |            ]
+         |          }
+         |        ]
+         |      }
+         |    ]
+         |  },
+         |  "limitSpec": {
+         |    "type": "default",
+         |    "limit": 10000,
+         |    "columns": [
+         |      {
+         |        "dimension": "userCount",
+         |        "direction": "descending"
+         |      }
+         |    ]
+         |  }
+         |}""".stripMargin.replaceAll("null", " ")
+    ProjectLogger.log("Druid Query Is " + druidQuery)
+    println("Query" + druidQuery)
+    val host: String = if (StringUtils.isNotBlank(ProjectUtil.getConfigValue("druid_proxy_api_host"))) ProjectUtil.getConfigValue("druid_proxy_api_host") else "localhost"
+    val port: String = if (StringUtils.isNotBlank(ProjectUtil.getConfigValue("druid_proxy_api_port"))) ProjectUtil.getConfigValue("druid_proxy_api_port") else "8081"
+    val endPoint: String = if (StringUtils.isNotBlank(ProjectUtil.getConfigValue("druid_proxy_api_endpoint"))) ProjectUtil.getConfigValue("druid_proxy_api_endpoint") else "/druid/v2/"
+    val request = Unirest.post(s"http://$host:$port$endPoint").headers(getUpdatedHeaders(new util.HashMap[String, String]())).body(druidQuery)
+    request.asString().getBody
+  }
+
+  def getCacheKey(batchId: String, intervals: String, groupByKeys: List[String]): String = {
+    val regex = "[^a-zA-Z0-9]"
+    val date = intervals.split("/")
+    s"bmetircs:$batchId:${date(0).replaceAll(regex, "")}:${date(1).replaceAll(regex, "")}:${groupByKeys.mkString(" ")}"
+  }
```

Comment:
Let us discuss with Santhosh about the key generation logic once.

Reply:
Sure